### PR TITLE
Add ibmgpu compiler for XL+OpenMP_offload runs on Summit

### PR DESCRIPTION
**Pull request by amametjanov**
Add ibmgpu compiler for XL+OpenMP_offload runs on Summit

[BFB]

### CODE REVIEW

**Comment by minxu74** [comment]
@amametjanov if the PR is ready,  I can merge it to next and master tomorrow. I just reviewed it and it is good to go.

File: cime/config/e3sm/machines/config_machines.xml [hunk 1]
```diff
@@ -2890,7 +2890,7 @@
     <DESC>ORNL Summit. Node: 2x POWER9 + 6x Volta V100, 22 cores/socket, 4 HW threads/core.</DESC>
     <NODENAME_REGEX>.*summit.*</NODENAME_REGEX>
     <OS>LINUX</OS>
-    <COMPILERS>ibm,pgi,pgigpu,pgiacc,gnu</COMPILERS>
+    <COMPILERS>ibm,ibmgpu,pgi,pgigpu,pgiacc,gnu</COMPILERS>
```
Comment (sarats) [inline cime/config/e3sm/machines/config_machines.xml:2893]: Probably remove pgiacc here.

**Comment by whannah1** [comment]
@amametjanov This looks like it might conflict with my PR to fix the summit config (#3616). Would it be better to just add these changes to that PR?

**Comment by minxu74** [comment]
@amametjanov there are some conflicts between your PR and next and master to which I just merged @whannah1 PR. Will you solve the conflicts on your branch? or I can solve them when I merge it into next and master? either way works for me. Thanks.

**Comment by amametjanov** [comment]
Resolved conflicts by rebasing onto latest master.

**Comment by rljacob** [comment]
This branch has been merged to master but github thinks it has not.   Will wait to see if it clears up.

**Pull request merged**